dronekit/cameras.py

```diff
@@ -0,0 +1,138 @@
+__author__ = 'anton'
+
+# Camera module
+"""
+This is a library for controlling cameras. It's a typical thing for companion computers to do,
+so it's a handy extensions of dronekit.
+
+The lib is primariliy meant for cams with wifi capability. But USB and Ir might be possible too.
+
+Creating an camera object supposes your computer is connected to the wifi hotspot of the camera you are trying to control
+
+----
+"""
+
+from urllib2 import urlopen
+from time import localtime
+import signal
+from . import errprinter
+import logging
+
+class Timeout():
+    """Timeout class using ALARM signal."""
+    class Timeout(Exception):
+        pass
+
+    def __init__(self, sec):
+        self.sec = sec
+
+    def __enter__(self):
+        signal.signal(signal.SIGALRM, self.raise_timeout)
+        signal.alarm(self.sec)
+
+    def __exit__(self, *args):
+        signal.alarm(0)    # disable alarm
+
+    def raise_timeout(self, *args):
+        raise Timeout.Timeout()
+
+
+class GeoTagger(object):
+    def __init__(self, vehicle=None):
+        self.vehicle = vehicle
+        FORMAT = '%(asctime)s, %(message)s'
+        DATEFORMAT = "%d-%m-%Y, %H:%M:%S"
+        logging.basicConfig(format=FORMAT, datefmt=DATEFORMAT, filename="camera_log.csv", filemode='w', level=logging.INFO)
+        logging.info("picture number, waypoint, gimbal pitch, gimbal yaw, gimbal roll, att pitch, att yaw, att roll, latitude, longitude, altitude, camera message")
+
+    def log_vehicle_state(self, num_picture, cam_message):
+        if self.vehicle:
+            log_msg = ",".join(map(str,[num_picture, self.vehicle.commands.next,
```

Comment:
I wonder if we should use the CSV lib for this, did you consider that when exploring this solution?